### PR TITLE
dcrdata/exchanges: connect to signalR websocket without signalR client

### LINKED ISSUE #1818: exchanges: need new signalr client
See https://github.com/decred/dcrdata/pull/1815#discussion_r614154297

Our signalr client repo is archived. https://github.com/rainhq/signalr is a candidate for replacement.

### PR DESCRIPTION
**Pull request by ukane-philemon**
This implements a `connectSignalRWebsocket` function which is used to connect to a `signalR` websocket without a signalR client. The connection is done in three steps (`negotiate`, `connect`, `start`)  and this makes it easy to connect to a `signalR` websocket without having to manage the `signalR` websocket connection separately from the regular websocket connection used by other exchanges. Closes #1818 

- Remove external dependency used to connect to a signalR websocket
- Fix an error in the exchanges live test.
- Update `bittrex` `signalR` websocket version.

For a quick test: 
```
$ go test --tags livexc -run TestBittrexLiveWebsocket                 
2022-12-08 14:03:50.987 [TRC] EXE: Initializing websocket connection for bittrex
2022-12-08 14:03:52.843 [TRC] EXE: Signalling candlestick update for bittrex, bin size 1d
2022-12-08 14:03:53.235 [TRC] EXE: Signalling candlestick update for bittrex, bin size 1h
Bittrex emitted an exchange update: 115 bids, 500 asks, 0 candlestick sets 
2022-12-08 14:03:59.164 [TRC] EXE: last bittrex websocket update 0.000 seconds ago
Bittrex emitted an exchange update: 115 bids, 500 asks, 2 candlestick sets 

```

### CODE REVIEW

File: exchanges/websocket.go [hunk 1]
```diff
@@ -94,115 +109,19 @@ func newSocketConnection(cfg *socketConfig) (websocketFeed, error) {
...
-	fmt.Println("=================================")
-	fmt.Printf("C: %s\n", jsonify(msg.C))
-	fmt.Printf("S: %s\n", jsonify(msg.S))
-	fmt.Printf("G: %s\n", jsonify(msg.G))
-	fmt.Printf("I: %s\n", jsonify(msg.I))
-	fmt.Printf("E: %s\n", jsonify(msg.E))
-	s, _ := msg.R.MarshalJSON()
-	fmt.Printf("R: %s\n", string(s))
-	s, _ = msg.H.MarshalJSON()
-	fmt.Printf("H: %s\n", string(s))
```
Comment (ukane-philemon) [inline exchanges/websocket.go:126]: I didn't see a use for this.

**Comment by chappjc** [comment]
Excellent work with this.  Really simplifies things as a bonus on top of eliminating the dependency.

File: exchanges/exchanges_test.go [hunk 2]
```diff
@@ -257,21 +256,6 @@ func (d *tDoer) queue(body interface{}) *http.Response {
...
-
-func (conn testBittrexConnection) On() bool {
-	// Doesn't matter right now.
-	return false
-}
-
-func (conn testBittrexConnection) Send(hubs.ClientMsg) error {
-	return nil
-}
-
```
Comment (ukane-philemon) [inline exchanges/exchanges_test.go:274]: Didn't see this used anywhere and tests run fine without it.

**Comment by chappjc** [comment]
Good work, but looks like trex was the only consumer of microsoft nonsense, so we'll close this this update.  I really do appreciate it though.

**Pull request closed without merging**